### PR TITLE
New Dokan admin notice design and Grouped notices.

### PR DESCRIPTION
**Pull request by shahadat015**
Testing Cases:

WooCommerce Missing Notice:
	Case 1: If WooCommerce doesn't exist show installation URL, 
	Case 2: But if exist but not activated show activation URL

Dokan Lite Missing Notice:
	Case 1: If Dokan Lite doesn't exist chek is showing install button and then install, 
	Case 2: If exist but not activated show activation URL

Limited-time promotions - Black Friday

-- Dokan Specific --
Setup wizard notice after installing Dokan Lite
If Dokan PRO doesn't exist show Dokan PRO promotional banner
Activate Dokan Pro Licence notice
Ask for review notice
Check What's new notice in Dokan Lite and PRO.
If the permalink settings structure is Plain and Numeric show notice to change permalink settings.
Dokan data update required:
	Case 1: Update data check it working or not...?

Geolocation Module notice - This notice will show after enabling Dokan Geolocation Module.
Booking Module notice - This notice will show after enabling Booking Module Module.
Elementor Module Notice - This notice will show after enabling Elementor Module
Stripe Connect Module Notice - This notice will show after enabling Stripe Connect Module.
Auction Integration Module Notice - This notice will show after enabling the Auction Integration Module.
Product Subscription module notice - This notice will show after the Product Subscription Module.
Product Addon Module Notice - This notice will show after enabling the Product Addon Module.
Table Rate Shipping Notice -  This notice will show after enabling the Table Rate Shipping Module.
Rank Math SEO Module notiece:
	Case 1: If the plugin does not exist show the Install button and click the install button
	Case 2: If exist show activation URL

Note: Check close and button actions working as expected?

Pro PR: https://github.com/weDevsOfficial/dokan-pro/pull/1588

### CODE REVIEW

File: src/admin/components/UpgradeBanner.vue [hunk 1]
```diff
@@ -1,23 +1,143 @@
...
     name: "UpgradeBanner",
+
     data() {
         return {
-            show: true,
-            upgradeURL: dokan.urls.upgradeToPro
+            show: dokan.showPromoBanner,
+            upgradeURL: dokan.urls.upgradeToPro,
+            features: [
+                this.__( 'Table rate shipping', 'dokan-lite' ),
```
Comment (nurul-umbhiya) [inline src/admin/components/UpgradeBanner.vue:135]: confirm this from content team

File: templates/admin-header.php [hunk 2]
```diff
@@ -5,60 +5,78 @@
...
-                rel="noopener"
-                title="<?php esc_attr_e( 'What\'s New', 'dokan-lite' ); ?>"
-            >
-                <?php echo esc_html_e( 'What\'s New', 'dokan-lite' ); ?>
-            </a>
-        <?php } ?>
+        <div class="menu-item">
+            <div class="item">
+                <?php
+                    $dokan_lite_versions        = get_option( 'dokan_lite_changelog_versions', array() );
```
Comment (nurul-umbhiya) [inline templates/admin-header.php:11]:
```
$cache_key = 'changelog_accessed_' . DOKAN_PLUGIN_VERSION;
$lite_viewed = Cache::get_transient( $cache_key );
```

**Review by nurul-umbhiya — CHANGES_REQUESTED** [review]

File: assets/src/less/global-admin.less [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+            .total-notice.active {
+                font-weight: 600;
+            }
+
+            .next,
+            .prev {
+                color: #dadfe4;
+            }
+
+            .next.active,
```
Comment (ManiruzzamanAkash) [inline assets/src/less/global-admin.less:290]: you can take this to top `.next, .prev`
```scss
.next,
.prev {
    color: #dadfe4;

    &.active {
       color: #9da6ae;
       cursor: pointer;
   }
}

File: includes/Admin/functions.php [hunk 4]
```diff
@@ -631,3 +634,136 @@ function showTooltip(x, y, contents) {
...
+ *
+ * @since 3.3.3
+ *
+ * @return array | void
+ */
+function dokan_get_admin_notices() {
+    $notices = apply_filters( 'dokan_admin_notices', [] );
+
+    if ( empty( $notices ) ) {
+        return;
```
Comment (nurul-umbhiya) [inline includes/Admin/functions.php:649]: return empty array from here

File: includes/Admin/functions.php [hunk 5]
```diff
@@ -631,3 +634,136 @@ function showTooltip(x, y, contents) {
...
+
+/**
+ * Dokan promotional notices
+ *
+ * @since 3.3.3
+ *
+ * @return array | void
+ */
+function dokan_get_promo_notices() {
+    $promos = Cache::get_transient( 'dokan_promo_notices' );
```
Comment (nurul-umbhiya) [inline includes/Admin/functions.php:684]: remove dokan_ prefix

File: includes/Admin/functions.php [hunk 6]
```diff
@@ -631,3 +634,136 @@ function showTooltip(x, y, contents) {
...
+        $promos           = wp_remote_retrieve_body( $response );
+
+        if ( is_wp_error( $response ) || $response['response']['code'] != 200 ) {
+            $promos = '[]';
+        }
+
+        Cache::set_transient( 'dokan_promo_notices', $promos, DAY_IN_SECONDS );
+    }
+
+    $promos                  = json_decode( $promos, true );
```
Comment (nurul-umbhiya) [inline includes/Admin/functions.php:698]: check for null value  here, if null return empty array

File: src/admin/components/UpgradeBanner.vue [hunk 7]
```diff
@@ -1,23 +1,143 @@
...
-        You're using <span>Dokan Lite</span>. To unlock more features, consider
-        <a target="_blank" rel="noopener" :href="upgradeURL">
-            Upgrading to Pro</a
-        >.
-
-        <div class="close-button" title="Dismiss the notice" @click="dismiss()">
-            &times;
+    <div class="dokan-promo-banner notice" v-if="show && showUpgrade">
+        <div class="thumbnail">
+            <svg viewBox="0 0 219 146" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (tanmayjay) [inline src/admin/components/UpgradeBanner.vue:4]: Move this `svg` file in the root assets folder and import that here if it has the possibility to be used in other places.

File: templates/admin-header.php [hunk 8]
```diff
@@ -1,64 +1,78 @@
...
+                            <div class="dokan-icon">
+                                <div class="icon-setup-wizard"></div>
+                            </div>
+                            <?php esc_html_e( 'Run Setup Wizard', 'dokan-lite' ); ?>
+                        </a>
+                    </div>
+                </div>
+            </div>
+        </div>
+        <!--<div class="menu-item">
```
Comment (tanmayjay) [inline templates/admin-header.php:70]: remove these codes if not needed.

File: templates/whats-new.php [hunk 9]
```diff
@@ -0,0 +1,2186 @@
...
+                    'description' => '',
+                ],
+            ],
+        ],
+    ],
+    [
+        'version'  => 'Version 3.2.15',
+        'released' => '2021-10-13',
+        'changes'  => [
+            'New Feature' => [
```
Comment (tanmayjay) [inline templates/whats-new.php:114]: It is not recommended to use uppercase letters and/or spaces in array keys. You could process it as needed while rendering on the frontend.
`'improvement'`, `'new_feature'`, these should be used as keys. And while rendering at frontend, you could convert them into `Improvement`, `New Feature`.

**Review by tanmayjay — CHANGES_REQUESTED** [review]

**Review by xaviranik — CHANGES_REQUESTED** [review]

File: src/admin/components/UpgradeBanner.vue (hunk 7 above)
Comment (shahadat015) [reply src/admin/components/UpgradeBanner.vue:4]: Have no possibility to reuse this svg

**Comment by shanjidah** [comment]
Issue No 1: Background colour should be visible on the full page.
How To Reproduce:
1. login As Admin.
2. Goto Dokan menu
3. Go to ?->what'menu

Here is the screenshot:
Screen Shot 2021-12-13 at 5 34 19 PM

**Comment by shahadat015** [comment]
> Issue No 1: Background colour should be visible on the full page. How To Reproduce:
> 
> 1. login As Admin.
> 2. Goto Dokan menu
> 3. Go to ?->what'menu
> 
> Here is the screenshot: Screen Shot 2021-12-13 at 5 34 19 PM

Fixed

**Review by nurul-umbhiya — APPROVED** [review]

**Comment by shanjidah** [comment]
Issue No 2: Need a label on Dokan Notice

Here is the screenshot:

Screen Shot 2021-12-14 at 3 45 40 PM

Screen Shot 2021-12-14 at 3 46 21 PM

**Comment by shanjidah** [comment]
Issue No 3: After clicking Activate plugin button Rank Math Seo is not activated.

**Here is the screenshot:**

Screen Shot 2021-12-14 at 4 31 27 PM

**Comment by shanjidah** [comment]
Issue No 4: Admin Group is not working from module page Store review to last one

**Comment by shanjidah** [comment]
Title; Notice showing below All Logs/ Reports menu.

**Here is the screenshot:**


Screen Shot 2021-12-15 at 12 48 34 PM

**Comment by shahadat015** [comment]
> Issue No 3: After clicking Activate plugin button Rank Math Seo is not activated.
> 
> **Here is the screenshot:**
> 
> Screen Shot 2021-12-14 at 5 00 34 PM

Resolved

**Comment by shahadat015** [comment]
> Title; Notice showing below All Logs/ Reports menu.
> 
> **Here is the screenshot:**
> 
> Screen Shot 2021-12-15 at 12 48 34 PM

Resloved

**Pull request merged**